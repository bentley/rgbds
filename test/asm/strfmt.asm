VAL EQUS STRFMT("Hello %s! I am %d years old today!", "world", $f)
PRINTT "{VAL}\n"

N = -42
PRINTT STRFMT("signed %010d == unsigned %010u\n", N, N)

N = 112
FMT EQUS "X"
PRINTT STRFMT("\tdb %#03{s:FMT} %% 26\t; %#03{FMT}\n", N, N % 26)

TEMPLATE EQUS "\"%s are %s\\n\""
PRINTT STRFMT(TEMPLATE, "roses", "red")
PRINTT STRFMT(TEMPLATE, "violets", "blue")
PRINTT STRFMT(TEMPLATE, "void", 0, "extra")

PRINTT STRCAT(STRFMT(STRFMT("%%%s.%d%s", "", 9, "f"), _PI), \
	STRFMT(" ~ %s\n", STRFMT("%s%x", "thr", 238)))

PRINTT STRFMT("%d eol %", 1)
PRINTT "\n"

PRINTT STRFMT("invalid %w spec\n", 42)

PRINTT STRFMT("one=%d two=%d three=%d\n", 1)
