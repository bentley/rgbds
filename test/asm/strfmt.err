warning: strfmt.asm(14): [-Wstring-format]
    Formatting number as type 's'
ERROR: strfmt.asm(14):
    STRFMT: 1 unformatted argument(s)
error: Assembly aborted (1 errors)!
