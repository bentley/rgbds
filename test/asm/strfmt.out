Hello world! I am 15 years old today!
signed -000000042 == unsigned 4294967254
	db $70 % 26	; $08
roses are red
violets are blue
void are 0
3.141586304 ~ three
